Makefile

```diff
@@ -36,111 +33,214 @@ TESTSRC := $(COMMON_TESTSRC) $(FUNCTIONAL_TESTSRC) $(UNIT_TESTSRC)
-# Resolves to a list: bin/unit/a_test bin/unit/b_test bin/unit/c_test ...
-UNIT_TESTBIN_SRC=$(filter %_test.c, $(UNIT_TESTSRC))
-UNIT_TESTBINS=$(UNIT_TESTBIN_SRC:$(TESTS_DIR)/%_test.c=$(BINDIR)/%_test)
-
 #*************************************************************#
-# CFLAGS, ETC
+# CFLAGS, LDFLAGS, ETC
 #
 
 INCLUDE = -I $(INCDIR) -I $(SRCDIR) -I $(SRCDIR)/platform_$(PLATFORM) -I $(TESTS_DIR)
 
-DEFAULT_CFLAGS += -D_GNU_SOURCE -ggdb3 -Wall -pthread -Wfatal-errors -Werror -Wvla
-DEFAULT_CFLAGS += -DXXH_STATIC_LINKING_ONLY -fPIC
-DEFAULT_CFLAGS += -DSPLINTERDB_PLATFORM_DIR=$(PLATFORM_DIR)
+# use += here, so that extra flags can be provided via the environment
+
+CFLAGS += -D_GNU_SOURCE -ggdb3 -Wall -pthread -Wfatal-errors -Werror -Wvla
+CFLAGS += -DXXH_STATIC_LINKING_ONLY -fPIC
+CFLAGS += -DSPLINTERDB_PLATFORM_DIR=$(PLATFORM_DIR)
 
 # track git ref in the built library
 GIT_VERSION := "$(shell git describe --abbrev=8 --dirty --always --tags)"
-DEFAULT_CFLAGS += -DGIT_VERSION=\"$(GIT_VERSION)\"
+CFLAGS += -DGIT_VERSION=\"$(GIT_VERSION)\"
 
 cpu_arch := $(shell uname -p)
 ifeq ($(cpu_arch),x86_64)
   # not supported on ARM64
-  DEFAULT_CFLAGS += -msse4.2 -mpopcnt
   CFLAGS += -march=native
 endif
 
-# use += here, so that extra flags can be provided via the environment
-DEFAULT_CFLAGS += $(LIBCONFIG_CFLAGS)
-DEFAULT_LDFLAGS += -ggdb3 -pthread
+LDFLAGS += -ggdb3 -pthread
+
+LIBS      = -lm -lpthread -laio -lxxhash
+DEPFLAGS  = -MMD -MP
 
-# ##########################################################################
-# To set sanitiziers, use environment variables, e.g.
-#   DEFAULT_CFLAGS="-fsanitize=address" DEFAULT_LDFLAGS="-fsanitize=address" make debug
+#*************************************************************#
+# Flags to select release vs debug builds, verbosity, etc.
 #
-# Note(s):
-#  - Address sanitizer builds: -fsanitize=address
-#     - Ctests will be silently skipped with clang builds. (Known issue.)
-#       Use gcc to build in Asan mode to run unit-tests.
-#     - Tests will run slow in address sanitizer builds.
+
+help::
+	@echo Environment variables controlling the build
+	@echo '  BUILD_DIR: Base name for build output (Default: "build").'
+	@echo '    $$(BUILD_DIR)/obj: object files'
+	@echo '    $$(BUILD_DIR)/lib: libraries'
+	@echo '    $$(BUILD_DIR)/bin: executables'
+	@echo '  Note: setting BUILD_MODE and other flags may further modify BUILD_DIR'
+	@echo
+
+ifndef BUILD_DIR
+BUILD_DIR := build
+endif
+
 #
-#  - Memory sanitizer builds: -fsanitize=memory
-#     - Builds will fail with gcc due to compiler error. Use clang instead.
-#     - Tests will run even slower in memory sanitizer builds.
+# Build mode
 #
-CFLAGS += $(DEFAULT_CFLAGS) -Ofast -flto
-LDFLAGS += $(DEFAULT_LDFLAGS) -Ofast -flto
-LIBS = -lm -lpthread -laio -lxxhash $(LIBCONFIG_LIBS)
+ifndef BUILD_MODE
+BUILD_MODE=release
+endif
 
-DEPFLAGS = -MMD -MT $@ -MP -MF $(OBJDIR)/$*.d
+ifeq "$(BUILD_MODE)" "debug"
+CFLAGS  += -DSPLINTER_DEBUG
```

Comment:
For readability, it will be better to indent the block under if / else / elseif. I tried with an indent of 3-spaces, and the scripts still seemed to work.

If you can do this indentation in all places where the if/elseif blocks are used, in this pass, that will be great. 

Otherwise, feel free to push this to a follow-on issue. (I'm collecting up some mop-up items to be done after this change merges to main.)